apps/docs/src/content/learn/advanced/webgpu.mdx

```diff
@@ -0,0 +1,245 @@
+  WebGPU is still an experimental browser api and at the time of writing has [limited availability
+  across major browsers](https://caniuse.com/?search=webgpu). For this reason, Three.js's webgpu
+  renderer fallbacks to webgl when webgpu is not available.
+</Tip>
+<Tip type="tip">
+  This same approach can be used to swap out the default renderer for any other custom renderer.
+</Tip>
+
+<Example path="renderers/WebGPU" />
+
+<small>
+  Adapted from [this Three.js
+  example](https://threejs.org/examples/?q=webgpu#webgpu_performance_renderbundle).
+</small>
+
+The WebGPU renderer doesn't immediately render. If the renderer you provide needs to delay renderering, you
+can defer rendering by initially setting the renderMode to `manual` like so:
+
+```svelte title="App.svelte"
+<script>
+  import { Canvas, T } from '@threlte/core@next'
+  import { WebGPURenderer } from 'three/webgpu'
+  let renderMode = $state('manual')
+</script>
+
+<Canvas
+  {renderMode}
+  createRenderer={(canvas) => {
+    const renderer = new WebGPURenderer({
+      canvas,
+      antialias: true,
+      forceWebGL: false
+    })
+    renderer.init().then(() => {
+      renderMode = 'on-demand'
+    })
+    return renderer
+  }}
+>
+  <Scene />
+</Canvas>
+```
+
+### Vite
+
+WebGPU uses top-level async to determine WebGPU compatibility. Vite
+will often throw an error when it detects this.
+
+To circumvent this issue, the following can be added to your Vite config.
+
+```js
+// vite.config.js
+optimizeDeps: {
+  esbuildOptions: {
+    target: 'esnext'
+  }
+},
+build: {
+  target: 'esnext'
+}
+```
+
+Alternatively,
+[`vite-plugin-top-level-await`](https://github.com/Menci/vite-plugin-top-level-await)
+can be used, although less success has been reported with this method.
+
+## TSL
+
+A question that comes up often in Three.js development is "How do I extend Three.js's materials?".
+External libraries such as [three-custom-shader-material](https://www.npmjs.com/package/three-custom-shader-material)
+use a find and replace solution to get this job done. Three.js has identified
+that it's not an ideal solution and recommends using the [Three.js Shading Language](https://github.com/mrdoob/three.js/wiki/Three.js-Shading-Language) or TSL for short.
+
+The example below is an adaptation of [this](https://threejs.org/examples/?q=tsl#webgpu_tsl_angular_slicing)
+Three.js example. There are many more [TSL examples](https://threejs.org/examples/?q=tsl)
+within Three.js that you can use or adapt for your project.
+
+<Example path="shaders/slice/main" />
+
+### Using the new classes
+
+By default `<T>` doesn't know about the new classes and will
+error on things like `<T.MeshPhysicalNodeMaterial />`. You have two options.
+
+1. Give `<T>` all the new classes from `three/webgpu` with the [`extend`](/docs/reference/core/t#extending-the-default-component-catalogue)
+   function. Adding all the classes will increase the bundle size of your application
+   but is easy and works.
```

Comment:
```suggestion
1. Provide `<T>` with all the new classes from `three/webgpu` with the [`extend`](/docs/reference/core/t#extending-the-default-component-catalogue)
   function. It's important to note that using a wildcard import for adding all the classes will increase the bundle size of your application because both `three` and `three/webgpu` will be imported in a non-tree-shakable way. However, this method is easy and works.
```